homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -0,0 +1,157 @@
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        self._ble_device = discovery_info.device
+        parsed = parse_advertisement_data(
+            discovery_info.device, discovery_info.advertisement
+        )
+        self._discovered_adv = parsed
+        self.context["title_placeholders"] = {
+            "name": name_from_discovery(self._discovered_adv),
+        }
+        return await self.async_step_init()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+        else:
+            current_addresses = self._async_current_ids()
+            for discovery_info in async_discovered_service_info(self.hass):
+                self._ble_device = discovery_info.device
+                address = discovery_info.address
+                if address in current_addresses or address in self._discovered_advs:
+                    continue
+                parsed = parse_advertisement_data(
+                    discovery_info.device, discovery_info.advertisement
+                )
+                if parsed:
+                    self._discovered_adv = parsed
+                    self._discovered_advs[address] = parsed
+
+        if not self._discovered_advs:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        if user_input is not None:
+            self._name = name_from_discovery(self._discovered_adv)
+            self._bdaddr = user_input[CONF_ADDRESS]
+            await self.async_set_unique_id(self._bdaddr, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+            return await self.async_step_link()
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ADDRESS): vol.In(
+                        {
+                            address: f"{parsed.data['local_name']} ({address})"
+                            for address, parsed in self._discovered_advs.items()
+                        }
+                    )
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_link(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Given a configured host, will ask the user to press the button to pair."""
+        errors: dict[str, str] = {}
+        token = randomid(32)
+        self._client = MicroBotApiClient(
```

Comment:
Should we really create a new client every time this method is called in the same flow?